### PR TITLE
Update Dotnet iOS Export Process

### PR DESCRIPTION
**Pull request by atlasapplications**
Resolves #100123

~~This updates the iOS export build process to allow for hybrid globalization introduced in .NET 9 while preserving .NET 8 settings. Also allows for custom ICU or no ICU.~~

UPDATE:

After reading more about how future versions of .NET will handle globalization APIs it's unnecessary to support hybrid globalization so this updates the iOS export build process to check if the ICU file is needed (such as with .NET 8) and if not (such as with .NET 9+) the ICU file is simply omitted.

### CODE REVIEW

**Comment by raulsntos** [comment]
I noticed this PR is still a draft. Are there any other changes you want to make?

**Comment by atlasapplications** [comment]
Thank you for reminding me. I hadn't gotten around to testing this yet. I tested it today and there are a few issues.

My understanding was slightly off with how globalization features are being used in future versions of .NET, which I think I understand more now based off this [documentation](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-hybrid-mode.md). Switching from a .NET 8 and 9 branch shows the differences.

It seems that .NET 8 by default uses full ICU data (`icudt.dat`) but can be changed to a hybrid mode which would produce a slimmed down `icudt_hybrid.dat` file. The copy functionality seemed to have been written when this default behavior produced that file. Unfortunately, by .NET 9, this functionality changed as no ICU data is generated at all on iOS (including tvOS and maccatalyst) because it's no longer needed as native APIs are used instead.

So, basically the build system just never supported hybrid globalization and now it's not necessary for iOS with .NET 9+ since native APIs are used.

So for this PR, should I change the scope to basically just check if it's .NET 8 (as that is now the minimum as of Godot 4.4), look for either `icudt.dat` or `icudt_hyrbid.dat`, which would be adding support for hybrid globalization? Then, for .NET 9+, just skip the copy entirely? Also would adding the potential for a custom ICU file seem important at all?

**Comment by raulsntos** [comment]
If hybrid globalization is going away in .NET 9 and we never supported it, I wouldn't add support now. We'll move to .NET 10 eventually so there's no need to add support for something that we'll be removing soon after.

I'm not sure what's the use case for custom ICU files, it doesn't seem there is a big demand for that so I would not add support for it either.

**Comment by atlasapplications** [comment]
Sounds good. I revised the code to reflect what we discussed so it's ready for review now.

**Review by raulsntos — APPROVED** [review]
The changes in this PR look good to me. But I don't have an iOS device so I can't test that it works as expected.

**Comment by Repiteo** [comment]
Thanks!
